internal/armada/server/submit.go

```diff
@@ -765,29 +804,45 @@ func (server *SubmitServer) createJobsObjects(request *api.JobSubmitRequest, own
 	}(server.compressorPool, context.Background(), compressor)
 	compressedOwnershipGroups, err := compress.CompressStringArray(ownershipGroups, compressor.(compress.Compressor))
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
 	jobs := make([]*api.Job, 0, len(request.JobRequestItems))
 
 	if request.JobSetId == "" {
-		return nil, errors.Errorf("[createJobs] job set not specified")
+		return nil, nil, errors.Errorf("[createJobs] job set not specified")
 	}
 
 	if request.Queue == "" {
-		return nil, errors.Errorf("[createJobs] queue not specified")
+		return nil, nil, errors.Errorf("[createJobs] queue not specified")
 	}
 
+	responseItems := make([]*api.JobSubmitResponseItem, 0, len(request.JobRequestItems))
 	for i, item := range request.JobRequestItems {
+		jobId := getUlid()
```

Comment:
This looks like a change in behaviour in that previously we used to return `nil` for the id when we rejected jobs but now we return a jobId.  I think this is good in that we now have an id that the user can give us to search logs etc, but does have a couple of downsides:

- Users may get confused if they see jobIds that don't appear anywhere else in the system (e.g. lookout)
- It's a mildly breaking interface change- e.g. if clients previously had code that did `if jobResponse.JobId == nil...`  this will now behave differently.

I think I'm mildly in favour of this change, but it might be good to get a second opinion- @severinson ?